appdatabase/database.go

```diff
@@ -198,11 +188,144 @@ func MigrateEnsUsernames(db *sql.DB) error {
 			continue
 		}
 
-		_, err = db.Exec(`INSERT INTO ens_usernames (username, chain_id) VALUES (?, ?)`, username, defaultChainID)
+		_, err = sqlTx.Exec(`INSERT INTO ens_usernames (username, chain_id) VALUES (?, ?)`, username, defaultChainID)
 		if err != nil {
 			log.Error("Migrating ens usernames: failed to insert username into new database", "ensUsername", username, "err", err.Error())
 		}
 	}
 
 	return nil
 }
+
+const (
+	batchSize = 1000
+)
+
+func migrateWalletJSONBlobs(sqlTx *sql.Tx) error {
+	var batchEntries [][]interface{}
+
+	// Extract useful information from the receipt blob and store it as sql interpretable
+	//
+	// Dropped storing postState because it was replaced by the status after EIP 658
+	// Dropped duplicating logs until we have a better way to store them in a
+	// more structured way. They can be found in transfers.receipt still
+	// Dropped tx_hash because it is redundant with transaction_details.hash
+	// Dropped the bloom filter because in SQLite is not possible to use it in an
+	// efficient manner
+	// Extract useful information from the tx blob
```

Comment:
Another fun fact: The `blk_hash` we use for `eth` transfers is wrong. It gets locally calculated using the transaction header, but it doesn't match the real one. That's the reason why the `blocks` table has a bunch of entries with the same `blk_number` but different `blk_hash`. The real block hash is the one that's in the receipt.

Reply:
Will extract and add the original has then also